gtp_wrapper.py

```diff
@@ -116,13 +117,39 @@ class CGOSPlayer(CGOSPlayerMixin, GtpInterface):
     pass
 
 
-def make_gtp_instance(read_file, verbosity=1, cgos_mode=False):
+class KGSPlayer(MCTSPlayer):
+    """ A wrapper for playing on KGS
+    Adds 'courtsey pass' -- if the opponent passes, we pass, regardless
+    of score or our opinion on the game. """
+    def __init__(self, **kwargs):
+        self.they_passed = False
+        super().__init__(**kwargs)
+
+    def get_move(self, color):
+        if self.they_passed:
+            return gtp.PASS
+        return super().get_move(color)
+
+    def make_move(self, color, vertex):
+        if vertex == gtp.PASS:
```

Comment:
This one's a bit weird - the GTP protocol first calls "get_move" and then echos your move back at you with "make_move". So "make_move" can either be your own move or your opponent's move.

This will also behave weirdly with minigo selfplay mode in Minigui, fwiw.

Reply:
ooh good catch.  We don't have any state about which color "we" are, though... hmm.  Sounds like KGS-mode should be explicit and fallback to standard MCTS otherwise.